### DISCUSSION TITLE
Refresh or reload page

### DISCUSSION THREAD

**Discussion by grusgrus59** [Salamek/chromium-kiosk#99] [General]
Good morning, 

Great project that you approach in the right way.

I am currently testing the integration of chromium-kiosk and if it corresponds to our use, we will sponsor you with great pleasure :).

I have a problem, the terminal went into screen saver during the night and we have now uninstalled xscreensave to prevent this from happening again.

The browser now displays "404 not found", our page (angular) is in localhost (via nginx) and present in /var/www/html/. How to force restart the qiosk browser or force the page to refresh automatically? (without reboot OS). 
I don't see any setting in the config.yml.

Sincerely.
Laurent.

**Comment by Salamek** [comment]
Hi, i'm glad that you are happy with my project!

If your browser displays 404 (that is considered a load error by qiosk) it should automatically reload when that URL becomes available (is URL correct? Is nginx running? Is nginx config path correct?).

Anyway if you want qiosk to reload the page there are two, well three ways how to do it:
1) The correct but complex way: Using [Websocket "remote" control](https://github.com/Salamek/qiosk/tree/master/.docs#remote-control), example CLI client here https://github.com/Salamek/chromium-kiosk/blob/master/examples/qiosk-control.py, calling `python qiosk-control.py setUrl https://example.com` should load that URL ergo "reload" it
2) The hacky way: changing `HOME_PAGE` in config.yml to different URL forces qiosk to load that URL (only when change is detected)  so you can set it to `https://example.com/?a=b` and after page is loaded back to `https://example.com`
3) The YOLO way: `killall -u chromium-kiosk`  This will kill all `chromium-kiosk` user processes forcing user session to reload and to start qiosk again

**Comment by Salamek** [comment]
@grusgrus59 let me know if this A was sufficient for your Q
